### ISSUE TITLE
Group errors under the same key

### ISSUE THREAD

**Issue by jgoux** [nearform/graphql-hooks#302]
Hello,

### Description

I'd like to have all the errors objects grouped.

### Suggested implementation

Instead of having a boolean for the `error` property, why not put `fetchError`, `httpError` and `graphQLErrors` in the `error`, and set it as `null` when there is no error ? Or a new `errors` object for compatibility?

So : 

```js
{
  errors: {
    fetchError,
    httpError,
    graphQLErrors
  }
}
```

The reason for that is to ease the handling of errors when executing mutations.

This : 
```js
  let handleSubmit = async (values, actions) => {
    let { error, fetchError, httpError, graphQLErrors } = await registerSample({
      variables: { input: values }
    })

    if (error) {
      snackbar.error(errorsToMessage({ fetchError, httpError, graphQLErrors }))
    } else {
      snackbar.success("Sample registered")
    }

    actions.setSubmitting(false)
  }
```

would be a bit better with the proposal : 
```js
  let handleSubmit = async (values, actions) => {
    let { errors } = await registerSample({
      variables: { input: values }
    })

    if (errors) {
      snackbar.error(errorsToMessage(errors))
    } else {
      snackbar.success("Sample registered")
    }

    actions.setSubmitting(false)
  }
```

I know it's not much but it's those little things that add up that makes a great API. 😄

**Comment by bmullan91** [comment]
@jgoux sorry for the delay - I agree, much nicer 👍

**Comment by galmail** [comment]
I've created a PR for this: https://github.com/nearform/graphql-hooks/pull/354

**Comment by bmullan91** [comment]
Fixed in #354 - available in `graphql-hooks@4.0.0` 🎉
